app/assets/javascripts/router.js

```diff
@@ -7,8 +7,10 @@
   of the above pages support.
 
   The router is initialized with a set of routes: a mapping of URL path templates
-  to route controller objects. Path templates can contain placeholders
-  (`/note/:id`) and optional segments (`/:type/:id(/history)`).
+  to route controller objects. Path templates can contain
+    placeholders (`/note/:id`),
+    scoped placeholders (`/type:node way relation/:id`) and
```

Comment:
The current syntax is similar to Rails routes:
- https://api.rubyonrails.org/classes/ActionDispatch/Routing/Mapper/Base.html#method-i-match
- https://guides.rubyonrails.org/routing.html#non-resourceful-routes

If we want to stick to that, we need something like [segment constraints](https://guides.rubyonrails.org/routing.html#segment-constraints).